frontend/src/metabase/dashboard/reducers.js

```diff
@@ -409,6 +399,15 @@ const loadingDashCards = handleActions(
         };
       },
     },
+    [SET_LOADING_DASHCARDS_COMPLETE]: {
+      next: state => {
+        return {
+          ...state,
+          loadingIds: [],
+          loadingStatus: "complete",
```

Comment:
This is needed to handle a race condition in line 449 of `fetchDashboardCardData` in `data-fetching.js`.
 
When navigating back to a tab that was already visited, the `Promise.all` will resolve immediately because the previous dashcard requests were cached. This causes a race condition where `FETCH_CARD_DATA` is dispatched before `FETCH_DASHBOARD_CARD_DATA`, thus we would be left with `loadingIds` and a `loadingStatus` of `running`.

Here we explicit set the `loadingIds` to an empty array and the status to `complete` to avoid an infinite loading state.